src/PIL/ImageShow.py

```diff
@@ -218,6 +218,20 @@ def get_command_ex(self, file, title=None, **options):
         return command, executable
 
 
+class IPythonViewer(Viewer):
+    def show_image(self, image, **options):
+        display(image)
+        return 1
+
+
+try:
+    from IPython.display import display
+except ImportError:
+    pass
+else:
+    register(IPythonViewer)
```

Comment:
Any reason to put `IPythonViewer` before the unix viewers but after the other Windows/macOS ones? I think it would be best to put this at the top (or use `register(..., order=0)`) to help consistency in IPython (e.g. when combined with tools that offer PDF exports, etc.).

Reply:
Yep, no reason. I've moved it up.